src/Sylius/Component/Core/Inventory/Updater/OnHandQuantityUpdater.php

```diff
@@ -0,0 +1,42 @@
+<?php
+
+/*
+ * This file is part of the Sylius package.
+ *
+ * (c) Paweł Jędrzejewski
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Sylius\Component\Core\Inventory\Updater;
+
+use Sylius\Component\Core\Model\OrderInterface;
+use Sylius\Component\Core\Model\OrderItemInterface;
+use Webmozart\Assert\Assert;
+
+/**
+ * @author Anna Walasek <anna.walasek@lakion.com>
+ */
+final class OnHandQuantityUpdater implements OnHandQuantityUpdaterInterface
+{
+    /**
+     * {@inheritdoc}
+     */
+    public function decrease(OrderInterface $order)
+    {
+        $orderItems = $order->getItems();
+        foreach ($orderItems as $orderItem) {
+            Assert::greaterThanEq($orderItem->getQuantity(), 0, 'Quantity of order items cannot be negative number.');
+
+            /** @var OrderItemInterface $orderItem*/
+            $variant = $orderItem->getVariant();
+
+            if ($variant->isTracked()) {
+                Assert::greaterThanEq(($variant->getOnHand() - $orderItem->getQuantity()), 0, 'Quantity of variant items on hand cannot be negative number.');
```

Comment:
The message should be something like `Not enough units to decrease the inventory of a variant "X"`